src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postponementrequest/PostponementRequestAboutToStartHandler.java

```diff
@@ -32,16 +32,17 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
         final CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
         final SscsCaseData sscsCaseData = caseDetails.getCaseData();
 
-        final Optional<Hearing> optionalHearing = emptyIfNull(sscsCaseData.getHearings()).stream()
-                .filter(h -> h.getValue().getHearingDateTime().isAfter(LocalDateTime.now()))
-                .distinct()
-                .findFirst();
-
         final PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(sscsCaseData);
 
         if (HearingRoute.GAPS.equals(sscsCaseData.getRegionalProcessingCenter().getHearingRoute())) {
             response.addError(POSTPONEMENTS_NOT_POSSIBLE_FOR_GAPS);
+            return response;
         }
+        final Optional<Hearing> optionalHearing = emptyIfNull(sscsCaseData.getHearings()).stream()
+                .filter(h -> h.getValue().getStart().isAfter(LocalDateTime.now()))
+                .distinct()
+                .findFirst();
```

Comment:
I would suggest just doing:
`Hearing hearing = sscsCaseData.getLatestHearing();
setPostponementRequest(hearing, sscsCaseData)`
instead

Reply:
I have looked at this have seen that getLatestHearing() sorts it by hearingid (which may not be created in order) , then datetime etc but we only care about date which i think the code i pushed does : i will change it to :
```
final Optional<Hearing> optionalHearing = emptyIfNull(sscsCaseData.getHearings()).stream()
        .filter(hearing -> LocalDateTime.now().isBefore(hearing.getValue().getStart()))
        .distinct()
        .findFirst();
```

which is cleaner as there is no need to check for nullable for a postponement

Reply:
Agreed. `getLatestHearing` is not equivalent to Eliandro's work and therefore cannot be used.

Reply:
It orders by hearing date, then by hearing id after. For this we want the current hearing, which is the latest, I recommend using getLatestHearing so that we have consistency on this with all our S&L work.